src/sentry/tasks/app_store_connect.py

```diff
@@ -4,232 +4,13 @@
-
-
-def create_difs_from_dsyms_zip(dsyms_zip: str, project: Project) -> None:
-    with sentry_sdk.start_span(op="dsym-difs", description="Extract difs dSYM zip"):
-        with open(dsyms_zip, "rb") as fp:
-            created = create_files_from_dif_zip(fp, project, accept_unknown=True)
-            for proj_debug_file in created:
-                logger.debug("Created %r for project %s", proj_debug_file, project.id)
-
-
-def get_or_create_persisted_build(
-    project: Project, config: appconnect.AppStoreConnectConfig, build: appconnect.BuildInfo
-) -> AppConnectBuild:
-    """Fetches the sentry-internal :class:`AppConnectBuild`.
-
-    The build corresponds to the :class:`appconnect.BuildInfo` as returned by the
-    AppStore Connect API. If no build exists yet, a new "pending" build is created.
-    """
-    try:
-        build_state = AppConnectBuild.objects.get(
-            project=project,
-            app_id=build.app_id,
-            platform=build.platform,
-            bundle_short_version=build.version,
-            bundle_version=build.build_number,
-        )
-    except AppConnectBuild.DoesNotExist:
-        build_state = AppConnectBuild(
-            project=project,
-            app_id=build.app_id,
-            bundle_id=config.bundleId,
-            platform=build.platform,
-            bundle_short_version=build.version,
-            bundle_version=build.build_number,
-            uploaded_to_appstore=build.uploaded_date,
-            first_seen=timezone.now(),
-            fetched=False,
-        )
-        build_state.save()
-    return build_state
-
-
-def process_builds(
-    project: Project,
-    config: appconnect.AppStoreConnectConfig,
-    to_process: list[appconnect.BuildInfo],
-) -> list[tuple[appconnect.BuildInfo, AppConnectBuild]]:
-    """Returns a list of builds whose dSYMs need to be updated or fetched.
-
-    This will create a new "pending" :class:`AppConnectBuild` for any :class:`appconnect.BuildInfo`
-    that cannot be found in the DB. These pending :class:`AppConnectBuild`s are immediately saved
-    upon creation.
-    """
-
-    pending_builds = []
-
-    with sentry_sdk.start_span(
-        op="appconnect-update-builds", description="Update AppStoreConnect builds in database"
-    ):
-        for build in to_process:
-            build_state = get_or_create_persisted_build(project, config, build)
-            if not build_state.fetched:
-                pending_builds.append((build, build_state))
-
-    LatestAppConnectBuildsCheck.objects.create_or_update(
-        project=project, source_id=config.id, values={"last_checked": timezone.now()}
-    )
-
-    return pending_builds
-
-
-# Untyped decorator would stop type-checking of entire function, split into an inner
-# function instead which can be type checked.
-@instrumented_task(
-    name="sentry.tasks.app_store_connect.refresh_all_builds",
-    queue="appstoreconnect",
-    ignore_result=True,
-)
-def refresh_all_builds() -> None:
```

Comment:
If there are any inflight tasks when these changes are deployed you'll get errors captured in sentry.

Reply:
Should I just keep it as noop? This function gets called by a cronjob, and I already removed the cronjob declaration?

Reply:
If you're confident there won't be any tasks in-flight when this is deployed removal is fine. Otherwise a no-op task will ensure you don't create an error.